suite/tests/api/ir_x86.c

```diff
@@ -222,28 +236,75 @@ test_all_opcodes_3(void *dc)
+{
+#    define INCLUDE_NAME "ir_x86_3args.h"
 #    include "ir_x86_all_opc.h"
-#    undef OPCODE_FOR_CREATE
-#    undef XOPCODE_FOR_CREATE
 #    undef INCLUDE_NAME
 }
 
 static void
 test_all_opcodes_3_avx(void *dc)
 {
 #    define INCLUDE_NAME "ir_x86_3args_avx.h"
-#    define OPCODE_FOR_CREATE(name, opc, icnm, flags, arg1, arg2, arg3)             \
-        do {                                                                        \
-            if ((flags & IF_X64_ELSE(X86_ONLY, X64_ONLY)) == 0) {                   \
-                instrlist_append(ilist, INSTR_CREATE_##icnm(dc, arg1, arg2, arg3)); \
-                len_##name = instr_length(dc, instrlist_last(ilist));               \
-            }                                                                       \
-        } while (0);
 #    include "ir_x86_all_opc.h"
+#    undef INCLUDE_NAME
+}
+
+static void
+test_all_opcodes_3_avx512_vex(void *dc)
+{
+#    define INCLUDE_NAME "ir_x86_3args_avx512_vex.h"
+#    include "ir_x86_all_opc.h"
+#    undef INCLUDE_NAME
+}
+
+static void
+test_all_opcodes_mask_3_avx512_evex(void *dc)
+{
+#    define INCLUDE_NAME "ir_x86_mask_3args_avx512_evex.h"
+#    include "ir_x86_all_opc.h"
+#    undef INCLUDE_NAME
+}
+
 #    undef OPCODE_FOR_CREATE
+#    undef XOPCODE_FOR_CREATE
+
+/***************************************************************************/
+/*********************** OPCODE_FOR_CREATE 4 args **************************/
+
+#    define OPCODE_FOR_CREATE(name, opc, icnm, flags, arg1, arg2, arg3, arg4)      \
+        do {                                                                       \
+            if ((flags & IF_X64_ELSE(X86_ONLY, X64_ONLY)) == 0) {                  \
+                instrlist_append(ilist,                                            \
+                                 INSTR_CREATE_##icnm(dc, arg1, arg2, arg3, arg4)); \
+                len_##name = instr_length(dc, instrlist_last(ilist));              \
+            }                                                                      \
+        } while (0);
+#    define XOPCODE_FOR_CREATE(name, opc, icnm, flags, arg1, arg2, arg3, arg4)     \
+        do {                                                                       \
+            if ((flags & IF_X64_ELSE(X86_ONLY, X64_ONLY)) == 0) {                  \
+                instrlist_append(ilist,                                            \
+                                 XINST_CREATE_##icnm(dc, arg1, arg2, arg3, arg4)); \
+                len_##name = instr_length(dc, instrlist_last(ilist));              \
+            }                                                                      \
+        } while (0);
+
+static void
+test_all_opcodes_4(void *dc)
+{
+#    define INCLUDE_NAME "ir_x86_4args.h"
+#    include "ir_x86_all_opc.h"
 #    undef INCLUDE_NAME
 }
 
+#    undef OPCODE_FOR_CREATE
+#    undef XOPCODE_FOR_CREATE
+
+/***************************************************************************/
```

Comment:
Use the original trailing banner style which this PR deleted: it indicates closing while this flat line does not.

Reply:
Done.